src/stim/py/compiled_measurement_sampler_pybind_test.py

```diff
@@ -118,6 +119,37 @@ def test_skip_reference_sample():
         [[True]],
     )
 
+def test_reference_sample_init():
+    np.testing.assert_array_equal(
+        stim.Circuit("X 0\nM 0").compile_sampler(reference_sample=None).sample(1),
+        [[True]],
+    )
+    circuit = stim.Circuit("X 0\nM 0")
```

Comment:
Add a sanity check that specifying `circuit.reference_sample()` gives the same samples as specifying None, for a circuit with deterministic measurements.